### PR TITLE
Adds a Cyborg Remote Monitor program for modular computers.

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Adds a program that allows for monitoring borg status, cell charge, and upgrades given. It also allows for the user to send a message to the borg. Messaging is a one-way communication, and requires an ID entered into the device so that the message has a name attached. All messages are sent to a borg's AI as well, if one is synced.

## Why It's Good For The Game
More useful programs for tablets is nice, and this is something I've wanted for a while as robo. The message function serves as convenient way to get borgs to come back to the workshop for upgrades.

Borg AI syncing is intentionally not displayed in this program. That info flat out gives away emagged borgs, and is one of the two pieces of info that you can use to confirm a Malf AI, so I'm happy with it being locked away onto the borg console.

## Changelog
:cl:
add: A cyborg remote monitor program is now available for modular computer devices (including tablets).
/:cl:

Notes:
- The upgrade list displays the illegal weapons upgrade, which might take away from some of the possible confusion you get (since the upgrade adds the "The cover interface glitches out for a split second" emagged text). Possible solutions include simply not listing the upgrade, but that would make it a bit inconsistent for no obvious reason. I'm not entirely sure this is a big deal anyway. Real emags are not affected.
- ~~I'll probably want to add in a program icon, so that it's visible in the system tray when running in the background.~~ This is done.

Images of the UI with and without an ID.
image

### CODE REVIEW

**Comment by trollbreeder** [comment]
Based zxaber deletes the entire changelog accidentally due to merge conflicts

**Comment by stylemistake** [comment]
LGTM

**Comment by Pumpkinoe** [comment]
does illegal technology show up

**Comment by zxaber** [comment]
The illegal tech upgrade that is created from the ExoFab shows up. Emagging a borg will not show.

File: code/modules/modular_computers/file_system/programs/borg_monitor.dm [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		data["cyborgs"] += list(cyborg_data)
+	return data
+
+/datum/computer_file/program/borg_monitor/ui_act(action, params)
+	if(..())
+		return
+
+	switch(action)
+		if("messagebot")
+			var/mob/living/silicon/robot/R = locate(params["ref"]) in GLOB.silicon_mobs
```
Comment (optimumtact) [inline code/modules/modular_computers/file_system/programs/borg_monitor.dm:55]: you need to check it's type post locate

File: html/changelog.html [hunk 2]
```diff
@@ -51,6 +51,7 @@
 		
 		-->
 		<div class="commit sansserif">
+			<h2 class="date">22 March 2020</h2>
```
Comment (optimumtact) [inline html/changelog.html:54]: remove this change

File: code/modules/modular_computers/file_system/programs/borg_monitor.dm (hunk 1 above)
Comment (zxaber) [reply code/modules/modular_computers/file_system/programs/borg_monitor.dm:56]: done

File: html/changelog.html (hunk 2 above)
Comment (zxaber) [reply html/changelog.html]: done

**Pull request merged**